R/commits.R

```diff
@@ -1,69 +1,41 @@
-get_repo <- function() {
-  proj <- usethis::proj_get()
-  git2r::repository(proj)
-}
-
 with_repo <- function(code) {
   withr::with_dir(usethis::proj_get(), code)
 }
 
 get_top_level_commits_impl <- function(since) {
-  commit <- git2r::commits(get_repo(), time = FALSE, n = 1)[[1]]
+  commit <- gert::git_log(max = 1)
 
   if (!is.null(since)) {
-    since_commit <- git2r::lookup_commit(since)
-    ab <- git2r::ahead_behind(commit, since_commit)
-    if (ab[[2]] > 0) {
+    ab <- gert::git_ahead_behind(ref = commit$commit, upstream = since$commit)
+    if (ab$behind > 0) {
       abort(paste0(since, " not reachable from current HEAD."))
     }
+    utils::head(gert::git_log(), ab$behind + 1)
   } else {
     since_commit <- NULL
+    since_commit
   }
-
-  get_first_parent(commit, since_commit)
```

Comment:
Please restore this code.

Reply:
It is not in use anymore but I restored it.

Reply:
Can you please restore this code too?

Reply:
I don't know which code you mean. The code from this comment has already been restored.

Reply:
AFAIR `get_first_parent()` is obsolete because gert does things simpler and totally different.

Re-adding it would require passing a `commit` object from {git2r} in several places.
I stopped trying now now after re-adding {git2r} in four functions again because {git2r} only works with its own object classes.

I don't think all functions need to be replaced 1by1 or **even can be**. I remember that several functions existed which I could not replace 1by1.

I believe that the return of 

```r
  commit <- gert::git_log(max = 1)

  if (!is.null(since)) {
    ab <- gert::git_ahead_behind(ref = commit$commit, upstream = since$commit)
    if (ab$behind > 0) {
      abort(paste0(since, " not reachable from current HEAD."))
    }
    utils::head(gert::git_log(), ab$behind + 1)
```

returns the same information as the {git2r} impl before.
I am saying "I believe" because I tested this months ago.
I could of course be wrong here.
These are non-trivial niche git situations.

That said, R CMD check passes (including the vignette which mimics/covers most basic functionality), I've used this implementation in several packages for several months now so this change is somewhat tested in practice.

I am not sure what the best way is to proceed here - I've spent hours in this already and already more than 30 mins in this comment alone.
If this PR/change is still too unsafe for you and you do not want to dive into yourself (which is ofc completely understandable), I am also fine if this PR would be closed.
When I started this PR I did not imagine it could become such a time sink.

Reply:
I stand corrected -- `git_log()` really only fetches "first parent" commits.

Reply:
This means that `get_first_parent()` can indeed go away.